src/SprykerEco/Zed/AfterPay/Business/Payment/Mapper/OrderToRequestTransfer.php

```diff
@@ -168,6 +178,40 @@ protected function buildOrderWithItemsRequestTransfer(AfterPayCallTransfer $afte
 
         $orderRequestTransfer = $this->addExpensesToOrderRequestTransfer($orderRequestTransfer, $afterPayCallTransfer);
         $orderRequestTransfer = $this->addGiftcardItems($orderRequestTransfer, $afterPayCallTransfer);
+        $orderRequestTransfer = $this->addShippingFeeToOrderRequestTransfer($orderRequestTransfer, $afterPayCallTransfer);
+
+        return $orderRequestTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AfterPayRequestOrderTransfer $orderRequestTransfer
+     * @param \Generated\Shared\Transfer\AfterPayCallTransfer $afterPayCallTransfer
+     *
+     * @return \Generated\Shared\Transfer\AfterPayRequestOrderTransfer
+     */
+    protected function addShippingFeeToOrderRequestTransfer(
+        AfterPayRequestOrderTransfer $orderRequestTransfer,
+        AfterPayCallTransfer $afterPayCallTransfer
+    ): AfterPayRequestOrderTransfer {
+        $amount = (string)$this->moneyFacade
+            ->convertIntegerToDecimal($afterPayCallTransfer->getTotals()->getShipmentTotal());
+
+        $shippingTax = $afterPayCallTransfer->getTotals()->getTaxTotal()->getAmount() -
```

Comment:
It's wrong way to calculate a shippingTax.


/**
 * @see \Spryker\Shared\Shipment\ShipmentConfig::SHIPMENT_EXPENSE_TYPE.
 */
protected const SHIPMENT_EXPENSE_TYPE = 'SHIPMENT_EXPENSE_TYPE';
    
    
$quoteTransfer->getExpenses()

    /**
     * @param \ArrayObject|\Generated\Shared\Transfer\ExpenseTransfer[] $expenseTransfers
     *
     * @return int
     */
    protected function getShipmentTaxTotal(ArrayObject $expenseTransfers): int
    {
        $shipmentTaxTotal = 0;

        foreach ($expenseTransfers as $expenseTransfer) {
            if ($expenseTransfer->getType() !== static::SHIPMENT_EXPENSE_TYPE) {
                continue;
            }

            $shipmentTaxTotal += $expenseTransfer->getSumTaxAmount();
        }

        return $shipmentTaxTotal;
    }